src/shared/bootspec.c

```diff
@@ -1404,13 +1404,113 @@ static void boot_entry_file_list(
                 *ret_status = status;
 }
 
+static void print_addon(
+                BootEntryAddon *addon,
+                const char *addon_str) {
+
+        printf("  %s: %s\n", addon_str, addon->location);
+        printf("      cmdline: %s%s\n", special_glyph(SPECIAL_GLYPH_TREE_RIGHT), addon->cmdline);
+}
+
+static int print_cmdline(const BootEntry *e) {
+
+        _cleanup_free_ char *final_cmdline = NULL;
+
+        assert(e);
+
+        if (!strv_isempty(e->options)) {
+                _cleanup_free_ char *t = NULL, *t2 = NULL;
+                _cleanup_strv_free_ char **ts = NULL;
+
+                t = strv_join(e->options, " ");
+                if (!t)
+                        return log_oom();
+
+                ts = strv_split_newlines(t);
+                if (!ts)
+                        return log_oom();
+
+                t2 = strv_join(ts, "\n              ");
+                if (!t2)
+                        return log_oom();
+
+                printf("  defCmdline: %s\n", t2);
+                if (!strextend(&final_cmdline, t2, " "))
+                        return log_oom();
+        }
+
+        LIST_FOREACH(addon_list, addon, e->local_addon_list) {
+                /* Add space at the beginning of addon_str to align it correctly */
+                print_addon(addon, " localAddon");
+                if (!strextend(&final_cmdline, addon->cmdline, " "))
+                        return log_oom();
+        }
+
+        if (final_cmdline)
+                printf(" finalCmdline: %s\n", final_cmdline);
+
+        return 0;
+}
+
+static int json_addon(
+                BootEntryAddon *addon,
+                const char *addon_str,
+                JsonVariant **array) {
+
+        int r;
+
+        r = json_variant_append_arrayb(array,
+                        JSON_BUILD_OBJECT(
+                                JSON_BUILD_PAIR(addon_str, JSON_BUILD_STRING(addon->location)),
+                                JSON_BUILD_PAIR("cmdline", JSON_BUILD_STRING(addon->cmdline))));
+        if (r < 0)
+                return log_oom();
+
+        return 0;
+}
+
+static int json_cmdline(const BootEntry *e, JsonVariant **v) {
+
```

Comment:
Also here. There should be no empty line between the function signature if it is one line, and the variable definitions.